src/Swift.Bindings/src/Parser/ModuleProcessor.cs

```diff
@@ -0,0 +1,223 @@
+        /// and all of its fields must also be from frozen types.
+        /// </summary>
+        /// <param name="structDecl">The struct declaration.</param>
+        /// <returns>True if the struct is frozen, false otherwise.</returns>
+        private bool EvaluateFrozenness(StructDecl structDecl)
+        {
+            if (!structDecl.IsFrozen)
+                return false;
+
+            // If any field is not frozen, the struct cannot be frozen.
+            foreach (var fieldDecl in structDecl.Fields)
+            {
+                if (fieldDecl.SwiftTypeSpec is not NamedTypeSpec namedFieldType)
+                    continue;
+
+                var fieldDatabaseRecord = _typeDatabase.Registrar.GetType(namedFieldType.Module, namedFieldType.NameWithoutModule)
+                    ?? throw new Exception($"Type not found in the database: {namedFieldType.NameWithoutModule}");
+
+                if (!fieldDatabaseRecord.IsFrozen)
+                    return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the struct can be treated as blittable under .NET's rules.
+        /// For now, uses Swift's ValueWitnessTable flags to detect non-POD or non-bitwise-takable.
+        /// </summary>
+        /// <param name="swiftTypeInfo">Struct's Swift type info containing the ValueWitnessTable pointer.</param>
+        /// <returns>True if blittable, false otherwise.</returns>
+        private unsafe bool EvaluateBlittability(SwiftTypeInfo swiftTypeInfo)
+        {
+            // TODO: Replace with a manual approach.
+            bool isNonPod = swiftTypeInfo.ValueWitnessTable->IsNonPOD;
+            bool isNonBitwiseTakable = swiftTypeInfo.ValueWitnessTable->IsNonBitwiseTakable;
+            return (!isNonPod && !isNonBitwiseTakable);
+        }
+
+        /// <summary>
+        /// Inserts a struct's details (e.g. name, metadata accessor, frozenness, blittability) into the type database.
+        /// </summary>
+        /// <param name="namedTypeSpec">Spec for the struct's name, module, etc.</param>
+        /// <param name="structDecl">The struct declaration node.</param>
+        /// <param name="swiftTypeInfo">Pointer plus ValueWitnessTable for the struct.</param>
+        /// <param name="isFrozen">Whether the struct is effectively frozen.</param>
+        /// <param name="isBlittable">Whether the struct is effectively blittable.</param>
+        private void RegisterStructType(NamedTypeSpec namedTypeSpec, StructDecl structDecl,
+            SwiftTypeInfo swiftTypeInfo, bool isFrozen, bool isBlittable)
+        {
+            _typeDatabase.Registrar.RegisterType(namedTypeSpec.Module,
+                                                 namedTypeSpec.NameWithoutModule,
+                                                 new TypeRecord
+                                                 {
+                                                     Namespace = namedTypeSpec.Module,
```

Comment:
I'd like to see a separate type for holding a type name.  Something like:
```
public class /* or record or struct */ TypeName {
     public string Namespace { get; init; }
     public string Name { get; init; }
     public string FullName { get; } => Namespace + "." + Name;
     // if not a record, make ToString, Equals, etc
}
```
This will make it easier to set up `import` statements in Swift and `using` statements in C#.

Reply:
I have a change ready, however it will affect multiple areas so Ill bring it in another PR